src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1559,12 +1739,57 @@ impl Cage {
         }
     }
 
+    /// ### Description
+    ///
```

Comment:
A line stating that this function is used by the read_syscall would be great

Reply:
fixed.